config/routes.rb

```diff
@@ -93,13 +93,16 @@
         end
       end
     end
-    resources :teams, except: [:edit, :update, :destroy] do
+    resources :teamsets, only: [:index, :edit, :update] do
       member do
-        patch :dissolve
-      end
-      collection do
-        patch :dissolve_all
+        patch :clone
         patch :randomize
+        patch :dissolve_all
+      end
+      resources :teams, only: [:show] do
```

Comment:
We should really work on unnesting these paths.

Best practice is max depth of 2, and even then only paths where the terminal object doesn't exist (e.g. index, new). Otherwise we end up with:

- Really long helper method names.
- Potential security issues, if users are providing ID#'s that aren't correct for intermediate path elements (like teamset / team).

Here I'd suggest not nesting teams under teamsets, and instead getting teamset_id from the team object.